builtin/providers/aws/resource_aws_codedeploy_deployment_group.go

```diff
@@ -362,3 +433,20 @@ func resourceAwsCodeDeployTagFilterHash(v interface{}) int {
 
 	return hashcode.String(buf.String())
 }
+
+func resourceAwsCodeDeployTriggerConfigHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%s-", m["trigger_name"].(string)))
+	buf.WriteString(fmt.Sprintf("%s-", m["trigger_target_arn"].(string)))
+	return hashcode.String(buf.String())
+}
+
+func validateTriggerEvent(v interface{}, k string) (ws []string, errors []error) {
+	value := v.(string)
+	if value != "DeploymentStart" && value != "DeploymentSuccess" && value != "DeploymentFailure" && value != "DeploymentStop" && value != "InstanceStart" && value != "InstanceSuccess" && value != "InstanceFailure" {
```

Comment:
This feels pretty hacky - this conditional will need to continue to be expanded if newer values are added. A potential pattern to use here may be something like this:

https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/validators.go#L81

This would mean the associated tests would then be able to be specified as follows:

https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_dynamodb_table_test.go#L58

rather than having 2 ranges within the test func